test-clients/src/main/java/com/hedera/services/bdd/suites/token/UniqueTokenManagementSpecs.java

```diff
@@ -377,43 +374,43 @@ private HapiApiSpec happyPathWithRepeatedMetadata() {
+                        cryptoCreate(FIRST_USER).balance(0L),
                         newKeyNamed("supplyKey"),
-                        newKeyNamed("signingKeyTreasury"),
-                        newKeyNamed("signingKeyFirstUser"),
-                        cryptoCreate(FIRST_USER).key("signingKeyFirstUser"),
-                        cryptoCreate(SECOND_USER),
-                        cryptoCreate(TOKEN_TREASURY).key("signingKeyTreasury"),
                         tokenCreate(A_TOKEN)
                                 .tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
                                 .supplyType(TokenSupplyType.INFINITE)
                                 .initialSupply(0)
                                 .supplyKey("supplyKey")
-                                .treasury(TOKEN_TREASURY),
-                        mintToken(A_TOKEN, List.of(ByteString.copyFromUtf8("memo"))),
-                        tokenAssociate(FIRST_USER, A_TOKEN),
-                        tokenAssociate(SECOND_USER, A_TOKEN)
-                ).when(
-                        cryptoTransfer(
-                                movingUnique(1, A_TOKEN).between(TOKEN_TREASURY, FIRST_USER),
-                                movingUnique(1, A_TOKEN).between(FIRST_USER, SECOND_USER)
-                        ).signedBy("signingKeyTreasury", "signingKeyFirstUser", DEFAULT_PAYER).via("cryptoTransferTxn")
-                ).then(
-                        getAccountBalance(TOKEN_TREASURY)
-                                .hasTokenBalance(A_TOKEN, 0),
-                        getAccountBalance(FIRST_USER)
-                                .hasTokenBalance(A_TOKEN, 0),
-                        getAccountBalance(SECOND_USER)
-                                .hasTokenBalance(A_TOKEN, 1),
-
-                        getTokenNftInfo(A_TOKEN, 1)
-                                .hasSerialNum(1)
-                                .hasMetadata(ByteString.copyFromUtf8("memo"))
-                                .hasTokenID(A_TOKEN)
-                                .hasAccountID(SECOND_USER),
-                        getTxnRecord("cryptoTransferTxn").logged()
+                                .treasury(TOKEN_TREASURY)
+                )
+                .when(
+                        mintToken(A_TOKEN, List.of(ByteString.copyFromUtf8("memo"))).via("mintTransferTxn")
+                )
+                .then(
+                        UtilVerbs.withOpContext((spec, opLog) -> {
+                            var mintNft = getTxnRecord("mintTransferTxn");
+                            allRunFor(spec, mintNft);
+                            var tokenTransferLists = mintNft.getResponseRecord().getTokenTransferListsList();
+                            tokenTransferLists.stream().forEach(tokenTransferList -> {
+                                Assert.assertNotNull(tokenTransferList.getNftTransfersList());
+                                tokenTransferList.getNftTransfersList().stream().forEach(nftTransfers -> {
```

Comment:
We need the test to be exact. If there is a bug and we produce 2 identical token transfer lists this test will pass. It will be better to verify that:
- the token transfer list size is correct
- the nft transfer size is correct
and then verify the exact `nftTransfer`

Reply:
Done